lib/config.js

```diff
@@ -702,6 +703,59 @@ util.resolveDeferredConfigs = function (config) {
     _iterate(config);
 }
 
+// Using basic recursion pattern, find all the shell-style read values and get file contents.
+util.shellStyleRead = function (config) {
+  // TODO (upon next major release):
+  //  process.env['NODE_CONFIG_SHELL_STYLE_READ'] === 'N'
+  if (process.env['NODE_CONFIG_SHELL_STYLE_READ'] !== 'Y') {
+    return;
+  }
+
+  var completeConfig = config;
+
+
+  function _iterate (prop) {
+
+    // We put the properties we are going to look at in an array to keep the order predictable
+    var propsToSort = [];
+
+    // First step is to put the properties of interest in an array
+    for (var property in prop) {
+      if (prop.hasOwnProperty(property) && prop[property] != null) {
+        propsToSort.push(property);
+      }
+    }
+
+    // Second step is to iterate of the elements in a predictable (sorted) order
+    propsToSort.sort().forEach(function (property) {
+      if (prop[property].constructor == Object) {
```

Comment:
What do you get by processing the entries in an order other than what they were defined in?

Seems like it could be hiding information from people trying to debug a failure during config loading